exec/java-exec/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -464,22 +464,30 @@ SqlNode SqlDropSchema(SqlParserPos pos) :
 
 /**
  * Parse refresh table metadata statement.
- * REFRESH TABLE METADATA tblname
+ * REFRESH TABLE METADATA [COLUMNS ((field1, field2,..) | NONE)] tblname
  */
 SqlNode SqlRefreshMetadata() :
 {
     SqlParserPos pos;
     SqlIdentifier tblName;
-    SqlNodeList fieldList;
+    SqlNodeList fieldList = null;
     SqlNode query;
+    boolean allColumns = true;
 }
 {
     <REFRESH> { pos = getPos(); }
     <TABLE>
     <METADATA>
+    [
+        <COLUMNS> { allColumns = false; }
+        (   fieldList = ParseRequiredFieldList("Table")
+            |
+            <NONE>
```

Comment:
Do we need `NONE` here? I think if columns keyword is present list of columns is required.

Reply:
Yes, since the columns keyword is optional, metadata will be collected for all the columns if the Columns keyword is not present. So the only way to express not to collect metadata for any column is to have NONE keyword which makes it more explicit.